File: 14. Longest Common Prefix/solution.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    let longest_prefix = "";
    let checks = 0;
    for (let i = 0; i < strs[0].length; i++) {
        let prefix = strs[0].substring(0, i + 1);
        strs.forEach((string) => {
            if (string.startsWith(prefix)) {
                checks += 1;
            } else {
                return longest_prefix;
            }
        })
        if (checks == (i + 1) * strs.length) {
            longest_prefix = prefix;
        }
    }
    return longest_prefix;
};

File: 14. Longest Common Prefix/solution.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        strs.sort()
        prefix = ""
        for i in range(len(strs[0])):
            if strs[0][i] != strs[-1][i]:
                break
            prefix += strs[0][i]
        return prefix